### DISCUSSION TITLE
Question: Is there any way we can use magit to show submodule status

### DISCUSSION THREAD

**Discussion by nqminhuit** [magit/magit#4703] [Support] [answered]
Current magit status only show for active repository, I am working on git submodule project with many repositories, every time I need to stage/commit, I have to navigate to the file changed then enable magit status and perform staging or committing. 

It would be nice if we can do this at the root module which show all files changed across all submodules.

**Comment by kyleam** [comment, marked as answer]
> every time I need to stage/commit, I have to navigate to the file changed then enable magit status and perform staging or committing.

With the default `diff.ignoreSubmodules`, an entry for a submodule should show up in the "Unstaged changes" section of the status buffer if the submodule has changes to the tracked file.  You can hit `RET` on this entry to visit the status buffer for the submodule.  (Also take a look at the [listing submodules](https://magit.vc/manual/magit/Listing-Submodules.html) node in the manual.)

> It would be nice if we can do this at the root module which show all files changed across all submodules.

Magit only supports showing a single entry for submodule (i.e it doesn't try to blur the boundaries between the repositories).  To see the details, you have to press `RET` on the entry to go to the status buffer for that repository.

**Comment by nqminhuit** [comment]
Hello,

Thank you for your response, but:
> an entry for a submodule should show up in the "Unstaged changes" section of the status buffer if the submodule has changes to the tracked file. You can hit RET on this entry to visit the status buffer for the submodule.

this feature does not show which files has changes on what module, especially when we have all modules committed changes and status buffer from root module will display all module have unstaged changes.

**Reply by kyleam** [reply]
> this feature does not show which files has changes on what module

Yes, my response wasn't claiming that it does.  I was pointing out that hitting `RET` on a module-level entry (whether it be something in the "Unstaged changes" section, an entry from shown via `magit-list-submodules`, or an entry in a section inserted via `magit-insert-modules`) sounds like a more direct route than what you described ("every time I need to stage/commit, I have to navigate to the file changed then enable magit status and perform staging or committing").

**Reply by nqminhuit** [reply]
If magit does not support this, then I guess I have to find another workaround. Thank you for your support!
